chapters/ch32.md

```diff
@@ -1,49 +1,47 @@
 
 > * If your training set is skewed towards one class, or if it has many classes, choose a “balanced” subset instead of 10 training examples at random out of the set of 100. For example, you can make sure that 2/10 of the examples are positive examples, and 8/10 are negative. More generally, you can make sure the fraction of examples from each class is as close as possible to the overall fraction in the original training set.
 
---> _replace THIS LINE by your translation for the above line_
+* Nếu tập huấn luyện của bạn bị lệch về một lớp, hoặc nếu nó có nhiều lớp, hãy chọn một tập hợp con “cân bằng” thay vì 10 mẫu huấn luyện ngẫu nhiên trong số 100 mẫu. Ví dụ, bạn có thể chắc chắn rằng 2/10 các mẫu là các mẫu dương và 8/10 là âm. Tổng quát hơn, bạn có thể đảm bảo tỷ lệ các mẫu trong mỗi lớp càng gần với tỉ lệ trong tập huấn luyện ban đầu.
 
 
 > I would not bother with either of these techniques unless you have already tried plotting learning curves and concluded that the curves are too noisy to see the underlying trends. If your training set is large—say over 10,000 examples—and your class distribution is not very skewed, you probably won’t need these techniques.
 
---> _replace THIS LINE by your translation for the above line_
+Nếu bạn đã thử vẽ các đồ thị quá trình học và kết luận rằng các đường cong quá nhiễu để nhìn thấy các xu hướng cơ bản, thì nên sử dụng một trong những kỹ thuật trên. Nếu tập huấn luyện của bạn có quy mô lớn, ví dụ hơn 10.000 mẫu, và phân phối lớp của bạn không bị lệch, có lẽ bạn không cần các kỹ thuật này.
 
 
 > Finally, plotting a learning curve may be computationally expensive: For example, you might have to train ten models with 1,000, then 2,000, all the way up to 10,000 examples. Training models with small datasets is much faster than training models with large datasets. Thus, instead of evenly spacing out the training set sizes on a linear scale as above, you might train models with 1,000, 2,000, 4,000, 6,000, and 10,000 examples. This should still give you a clear sense of the trends in the learning curves. Of course, this technique is relevant only if the computational cost of training all the additional models is significant.
 
---> _replace THIS LINE by your translation for the above line_
+Cuối cùng, vẽ đồ thị quá trình học có thể tốn kém về mặt tính toán: Ví dụ: bạn có thể phải huấn luyện mười mô hình với 1.000, rồi 2.000, cho đến 10.000 mẫu. Huấn luyện các mô hình với các bộ dữ liệu nhỏ nhanh hơn nhiều so với huấn luyện các mô hình với các bộ dữ liệu lớn. Do đó, thay vì cách đều các kích thước tập huấn luyện theo tỷ lệ tuyến tính như trên, bạn có thể huấn luyện các mô hình với 1.000, 2.000, 4.000, 6.000 và 10.000 mẫu. Điều này vẫn sẽ cung cấp cho bạn một cảm giác rõ ràng về các xu hướng trong các đồ thị quá trình học. Tất nhiên, kỹ thuật này chỉ thích hợp nếu chi phí tính toán để huấn luyện tất cả các mô hình bổ sung là đáng kể.
```

Comment:
-  tốn kém về mặt tính toán --> độ phức tạp tính toán cao 
- cảm giác --> cảm nhận

Reply:
cụm "độ phức tạp tính toán cao" học thuật hơn nhưng chắc chỉ phù hợp khi phân tích lý thuyết hay thuật toán; ở đây là ý nói là tốn công ngồi chờ & tốn tài nguyên để plot, mình vẫn prefer bản dịch hiện tại. 

"cảm giác" vs "cảm nhận" chắc ko khác nhau về nghĩa.

Reply:
Nếu là phân tích lý thuyết hay thuật toán thì mình hiểu là "độ phức tạp thuật toán". Đây chỉ là cách ghép từ lại cho mượt thôi, mình không rõ mọi người có hiểu theo ý khác nhiều không. Để nguyên cũng không vấn đề gì. 

> Cảm giác là một quá trình tâm lý phản ánh những thuộc tính riêng lẻ của các sự vật hiện tượng khi chúng trực tiếp tác động vào giác quan của loài người.

> Cảm nhận hay là Cảm quan là quá trình thu thập, nhận biết thông tin từ hệ giác quan

Cảm nhận mang ý rộng hơn một chút, còn cảm giác chỉ thông qua các loại giác quan thôi.  

Mà cái này cũng không quan trọng lắm.